### PR TITLE
SMA ramping

### PR DESCRIPTION
**Pull request by sporejack**
# Issue
#299 

# Changes
 - Add `desiredPeriods` instance variable to `SMAOracle`
 - Add guard to `SMAOracle::update`
 - Add explanatory comment to `SMAOracle`

### CODE REVIEW

**Comment by sporejack** [comment]
@mynameuhh Tests are failing due to one of the RV fixes not being present, which is unfortunate. My gut instinct is to disable the offending tests for now and then re-enable them when we do our coverage sweep?

**Comment by mynameuhh** [comment]
@sporejack Yep that sounds good to me.

Make a ticket and chuck it in the "In scope v2" column just to make sure we don't forget

File: contracts/implementation/SMAOracle.sol [hunk 1]
```diff
@@ -23,6 +71,10 @@ contract SMAOracle is IOracleWrapper {
     /// Number of periods to use in calculating the SMA (`k` in the SMA equation)
     uint256 public periods;
 
+    /// Number of desired sampling periods to use -- this will differ from
+    /// `periods` initially until the SMA oracle ramps up
+    uint256 desiredPeriods;
```
Comment (mynameuhh) [inline contracts/implementation/SMAOracle.sol:76]:
```suggestion
    uint256 immutable desiredPeriods;
```

**Review by mynameuhh — CHANGES_REQUESTED** [review]

File: contracts/implementation/SMAOracle.sol (hunk 1 above)
Comment (sporejack) [reply contracts/implementation/SMAOracle.sol]: Good call!

**Review by mynameuhh — APPROVED** [review]

**Review by 0xhrithik — APPROVED** [review]

**Pull request merged**